src/main/java/de/tum/in/www1/artemis/web/rest/lecture/AttachmentUnitResource.java

```diff
@@ -161,30 +164,50 @@ public ResponseEntity<AttachmentUnit> createAttachmentUnit(@PathVariable Long le
     }
 
     /**
-     * POST lectures/:lectureId/attachment-units/split : creates new attachment units. The provided file must be a pdf file.
+     * POST lectures/:lectureId/process-units/upload : Temporarily uploads a file which will be processed into lecture units
      *
-     * @param lectureId                 the id of the lecture to which the attachment units should be added
-     * @param lectureUnitInformationDTO the units that should be created
-     * @param file                      the file to be splitted
-     * @return the ResponseEntity with status 200 (ok) and with body the newly created attachment units
+     * @param file      the file that will be processed
+     * @param lectureId the id of the lecture to which the attachment units will be added
+     * @return the ResponseEntity with status 200 (ok) and with body filename of the uploaded file
      */
-    @PostMapping("lectures/{lectureId}/attachment-units/split")
+    @PostMapping("lectures/{lectureId}/process-units/upload")
     @EnforceAtLeastEditor
-    public ResponseEntity<List<AttachmentUnit>> createAttachmentUnits(@PathVariable Long lectureId, @RequestPart LectureUnitInformationDTO lectureUnitInformationDTO,
-            @RequestPart MultipartFile file) {
-        log.debug("REST request to create AttachmentUnits {} with lectureId {}", lectureUnitInformationDTO, lectureId);
-
-        Lecture lecture = lectureRepository.findByIdWithLectureUnitsElseThrow(lectureId);
-        if (lecture.getCourse() == null) {
-            throw new ConflictException("Specified lecture is not part of a course", "AttachmentUnit", "courseMissing");
+    public ResponseEntity<String> uploadSlidesForProcessing(@PathVariable Long lectureId, @RequestPart("file") MultipartFile file) {
+        // time until the temporary file gets deleted. Must be greater or equal than MINUTES_UNTIL_DELETION in attachment-units.component.ts
+        int minutesUntilDeletion = 30;
+        log.debug("REST request to upload file: {}", file.getOriginalFilename());
+        checkLecture(lectureId);
+        if (!Objects.equals(FilenameUtils.getExtension(file.getOriginalFilename()), "pdf")) {
+            throw new BadRequestAlertException("The file must be a pdf", ENTITY_NAME, "wrongFileType");
         }
-        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, lecture.getCourse(), null);
+
+        URI fileURI = fileService.handleSaveFile(file, false, false);
+        fileService.schedulePathForDeletion(FilePathService.actualPathForPublicPath(fileURI), minutesUntilDeletion);
+        String fileName = FilePathService.actualPathForPublicPath(fileURI).getFileName().toString();
+
+        return ResponseEntity.ok().body(GSON.toJson(fileName));
```

Comment:
Why do we return the actual path here? Should this not be the public path? Also the call of `toJson()` seems unnecessary to me.

Reply:
1. I return the filename - not the complete path (so example.pdf instead of .../temp/img/example.pdf). The idea is that further requests (e.g. createAttachmentUnits in line 201) can access the temporary file easily and also can **only** access files in the temp folder as i make the filename part of the URI. If you have an idea how to do this better please share it! :) (I'm also not 100% happy with this solution but it's the best I can see) 

2. toJson is necessary, as Spring Boot has an issue that returning strings breaks the response body.